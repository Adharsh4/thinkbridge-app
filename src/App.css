.app {
  min-height: 100vh;
  background-color: rgb(245, 243, 239);
  padding-bottom: 2%;
}

