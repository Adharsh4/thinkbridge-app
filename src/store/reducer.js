import * as actionTypes from './actionTypes';

const initialState = {
    products: [],
    isLoading: true,
    errorMessage: ""
}

const reducer = (state = initialState, action) => {
    switch(action.type){
        case actionTypes.START_FETCHING:
            return {
                ...state,
                isLoading: true,
                errorMessage: ""
            }
        case actionTypes.STOP_FETCHING:
            return {
                ...state,
                isLoading: false,
                errorMessage: ""
            }
        case actionTypes.SAVE_PRODUCTS:
            return {
                ...state,
                products: action.products,
                isLoading: false,
                selectedProduct: [],
                errorMessage: ""
            }
        case actionTypes.CHANGE_ISLOADING:
            return {
                ...state,
                isLoading: false,
                errorMessage: ""
            }
        case actionTypes.CHANGE_ISLOADINGTOTRUE:
            return {
                ...state,
                isLoading: true,
            }
        case actionTypes.SET_ERRORMESSAGE:
            return {
                ...state,
                errorMessage: action.message,
                isLoading: false
            }
        default:
            return state;
    }
}

export default reducer;