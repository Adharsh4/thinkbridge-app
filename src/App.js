
import './App.css';
import Login from './components/Login';
import Header from './components/Header';
import Inventry from './components/Inventry';
import ShowProducts from './components/ShowProducts';
import { Switch, Route, withRouter } from 'react-router-dom';

function App() {
  return (
    <div className="app">
      <Switch>
        <Route path="/showProducts">
            <Header />
            <ShowProducts />
        </Route>
        <Route path="/inventry">
            <Header />
            <Inventry />
        </Route>
        <Route path="/">
        <Header />
          <Login />
        </Route>
      </Switch>
 </div>
  );
}

export default withRouter(App);
