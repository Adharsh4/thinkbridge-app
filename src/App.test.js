
import React from 'react';
import App from "./App"
import Enzyme, {shallow, mount} from 'enzyme'
import EnzymeAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import {BrowserRouter} from 'react-router-dom'
import { createStore, applyMiddleware } from 'redux';
import reducer from "./store/reducer"
import thunk from 'redux-thunk';

Enzyme.configure({adapter: new EnzymeAdapter()})

describe('<App /> unit test', () => {
    const mockStore = createStore(reducer, applyMiddleware(thunk));
    const getWrapper = () => mount(
      <Provider store={mockStore}>
          <BrowserRouter>
            <App/>
        </BrowserRouter>
      </Provider>
    );
    const container = getWrapper();

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

  
  });
