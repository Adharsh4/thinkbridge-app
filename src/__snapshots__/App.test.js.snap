// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> unit test should match the snapshot 1`] = `"<div class=\\"app\\"><div class=\\"header-main\\"><div><a href=\\"/showProducts\\"><img class=\\"header__logo\\" src=\\"logo.png\\"></a></div><div class=\\"header__search\\"><input class=\\"header__searchInput\\" type=\\"text\\" name=\\"search\\" value=\\"\\"><svg class=\\"MuiSvgIcon-root header__searchIcon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></div><div class=\\"header__nav\\"><a class=\\"no-decoration\\" href=\\"/login\\"><div class=\\"header__option\\"><span class=\\"header__optionLineOne\\">Hello Guest</span><span class=\\"header__optionLineTwo\\">Sign In</span></div></a><a class=\\"no-decoration\\" href=\\"/showProducts\\"><div class=\\"header__option\\"><span class=\\"header__optionLineOne\\">Your</span><span class=\\"header__optionLineTwo\\">Products</span></div></a></div></div><div class=\\"login__container\\"><h1 class=\\"text-center\\">Sign-in</h1><form class=\\"form__alingment\\"><div class=\\"form-group\\"><input type=\\"email\\" class=\\"form-control\\" id=\\"email\\" placeholder=\\"Enter email\\" value=\\"\\"></div><div class=\\"form-group\\"><input type=\\"password\\" class=\\"form-control\\" id=\\"password\\" placeholder=\\"Password\\" value=\\"\\"></div><div class=\\"text-center\\"><button type=\\"submit\\" class=\\"login__signInButton\\">Sign In</button><button class=\\"custom-button\\">Switch to sign up</button></div></form></div></div>"`;
